src/metabase/search/postgres/index.clj

```diff
@@ -145,6 +145,16 @@
     (when @reindexing?
       (upsert! pending-table entry))))
 
+(defn delete!
+  "Remove any entries corresponding directly to a given model instance."
+  [id search-models]
+  ;; In practice, we expect this to be 1-1, but the data model does not preclude it.
+  (when (seq search-models)
```

Comment:
in which case entries that has the same id but different model?

Reply:
Well the models aren't 1-1, so when a card is updated we are going to search for "card" (question), "dataset" (model), and "metric" search-models. In practice there will only be one of these in the database at least.